sentry/src/main/java/io/sentry/clientreport/AtomicClientReportStorage.java

```diff
@@ -14,25 +16,28 @@
 @ApiStatus.Internal
 final class AtomicClientReportStorage implements IClientReportStorage {
 
-  private final @NotNull Map<ClientReportKey, AtomicLong> lostEventCounts;
+  private final @NotNull LazyEvaluator<Map<ClientReportKey, AtomicLong>> lostEventCounts =
+      new LazyEvaluator<>(
+          () -> {
+            final Map<ClientReportKey, AtomicLong> modifyableEventCountsForInit =
+                new ConcurrentHashMap<>();
 
-  public AtomicClientReportStorage() {
-    final Map<ClientReportKey, AtomicLong> modifyableEventCountsForInit = new ConcurrentHashMap<>();
+            for (final DiscardReason discardReason : DiscardReason.values()) {
```

Comment:
@adinauer do you remember a specific reason for creating this map on init here, rather than adding entries on-demand within `void addCount`? I see it makes `addCount` fast, but I'm wondering if this even required, as this shouldn't be called very often.

Reply:
IIRC I built it this way due to limited functionality of `ConcurrentHashMap` on older versions of Android. The way it is built now requires little and very specific synchronization at the cost of pre-initializing the map.

Here's some more details: https://github.com/getsentry/sentry-java/pull/1982/files/fff83e9b830bef077c21fc1e09ecfbf4fcef1a9c#r844882927

Reply:
Yeah makes a lot of sense! But `.addCount()` is only called in error cases and bound to network responses. So maybe it's better to have a slower `.addCount()` in favor of faster ctor?
I'm fine with LazyEvaluator too, just wondering what's best 😅

Reply:
It's also called for sampling, backpressure, event processor etc. which might happen a lot and is not necessarily bound to network or other slow things. I prefer using `LazyEvaluator`.